### ISSUE TITLE
Client overwriting my app's logback configuration

### ISSUE THREAD

**Issue by taylorKonigsmark** [kubernetes-client/java#276]
Hello everyone!

When adding the client as a dependency to my spring boot 2 application, the library is overwriting the logback configurations that I specified at /src/main/resources/logback-spring.xml.

my dependency config with gradle looks like:

    compile('org.springframework.boot:spring-boot-starter-web')
    compile('org.springframework.boot:spring-boot-starter-actuator')
    compile('org.springframework.cloud:spring-cloud-starter-consul-config')

    compile('org.elasticsearch.client:elasticsearch-rest-high-level-client:6.2.4')
    compile('org.elasticsearch.client:transport:6.2.4')
    compile ('io.reactivex.rxjava2:rxjava:2.1.13')
    compile ('io.kubernetes:client-java:1.0.0')
    compileOnly('org.springframework.boot:spring-boot-configuration-processor')
    compileOnly('org.projectlombok:lombok')
    testCompile('org.springframework.boot:spring-boot-starter-test')

my logback-spring.xml looks like this:
```
<configuration>
    <springProperty name="APP_VERSION" source="spring.application.version"/>
    <springProperty name="APP_NAME" source="spring.application.name"/>
    <appender name="consoleAppenderJson" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LoggingEventCompositeJsonEncoder">
            <providers>
                <mdc/><!-- MDC variables on the Thread will be written as JSON fields -->
                <context/> <!-- Outputs entries from logback's context -->
                <logLevel/>
                <loggerName/>
                <pattern>
                    <pattern>
                        {
                        "appName": "${APP_NAME}",
                        "appVersion": "${APP_VERSION}"
                        }
                    </pattern>
                </pattern>
                <threadName/>
                <message/>
                <logstashMarkers/> <!-- Useful so we can add extra information for specific log lines as Markers -->
                <arguments/> <!-- or through StructuredArguments -->
                <stackTrace/>
            </providers>
        </encoder>
    </appender>
    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
            <pattern>%date{yyyy-MM-dd HH:mm:ss.SSS} %-4relative [%thread] %-5level %marker %logger{35} - %msg %n
            </pattern>
        </encoder>
    </appender>
    <springProfile name="default,dev,ci,perf,stg,uat,prod">
        <root>
            <appender-ref ref="consoleAppenderJson"/>
        </root>
    </springProfile>
    <springProfile name="local">
        <root>
            <level value="INFO"/>
            <appender-ref ref="STDOUT"/>
        </root>
    </springProfile>
</configuration>
```

I couldn't find any documentation on this, so I imagine that this shouldn't be happening.

**Comment by brendandburns** [comment]
I can't imagine any reason why this would happen, unless somehow we're linking a spring-boot library accidentally?  But I don't think so.  I don't see any relevant libraries in a quick survey of our pom.xml files...

**Comment by taylorKonigsmark** [comment]
Thanks for the reply!

There is one other dependency that isn't in this picture, it's a internal library used by the company I'm working for. I figured it's not part of the issue as I would think these two dependencies wouldn't be conflicting with each other.

Here are the dependencies of that internal library for more visibility:

<img width="590" alt="screen shot 2018-05-25 at 8 46 58 am" src="https://user-images.githubusercontent.com/16637257/40547989-066bf442-5ff9-11e8-9d30-73edd6d44607.png">

While this internal library does have spring boot packages in there. I excluded them in gradle resulting in this:

<img width="547" alt="screen shot 2018-05-25 at 8 47 55 am" src="https://user-images.githubusercontent.com/16637257/40548073-47dc32a2-5ff9-11e8-84f7-2000a72e3913.png">

And the story is still the same, this dependency overwrites my logback-spring configuration.

**Comment by brendandburns** [comment]
Is there an easy way for me to try to reproduce this? I don't really do much with Spring...

Thanks

**Comment by taylorKonigsmark** [comment]
Going to close the ticket. Tried to replicate it and couldn't, it's probably something else clashing with the dependencies of this library.

Sorry about the confusion!

**Comment by remmeier** [comment]
I get:

```
 [jar:file:/C:/Users/Remo/.gradle/caches/modules-2/files-2.1/io.kubernetes/client-java/1.0.0/4bf42473e37aab25ec95ba6095dd26ea6a132c01/client-java-1.0.0.jar!/logback.xml]
```

and when looking at pom.xml in META-INF/maven/io.kubernetes/client-java I see:

```
    <dependency>
            <groupId>ch.qos.logback</groupId>
            <artifactId>logback-classic</artifactId>
            <version>1.2.3</version>
     </dependency>
```

both should go away, it is not the responsiblity of a library to provide the logging setup and a bit cumbersome to get rid of again in the application.

**Comment by remmeier** [comment]
something like the attached commit should do, can check later if it works properly

**Comment by brendandburns** [comment]
Fixed by #284
